Azure-Samples/chat-with-your-data-solution-accelerator
# feat: Chat History and cosmosdb implementation

## Purpose

This pull request introduces the addition of chat history to the Chat With Your Data (CWYD) website, leveraging Cosmos DB for persistent storage of user conversations. This allows users to access past interactions, making the platform more user-friendly and improving the continuity of conversations. It also helps to refine the AI responses by referencing previous queries.

## Does this introduce a breaking change?





## How to Test
*  Get the code
- Download this PR and install using command line azd up
- go to the CWYD new frontend enviroment 
- Check the new Chat history features
-
```
git clone [repo-address]
cd [repo-name]
git checkout [branch-name]
npm install
```

* Test the code

```
```

## What to Check
Verify that the following are valid
- Chat history is stored correctly in Cosmos DB.
- Users can retrieve and view past chats.
- Performance remains consistent after chat history is enabled.

## Other Information

This update integrates smoothly with existing infrastructure without altering the core chat functionalities, allowing for a seamless experience for both users and administrators.
